### PR TITLE
[FLINK-11296][Table API & SQL] Support truncate in TableAPI

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
## What is the purpose of the change

*This PR   add `truncate` as core scalar operators in then codegen.*

## Brief change log

*(for example:)*

- docs/dev/table/functions.md
- table/api/scala/expressionDsl.scala
- table/codegen/calls/BuiltInMethods.scala
- table/codegen/calls/FunctionGenerator.scala
- table/expressions/mathExpressions.scala
- table/validate/FunctionCatalog.scala
- table/expressions/ScalarFunctionsTest.scala
- table/expressions/SqlExpressionTest.scala

## Verifying this change

*(Please pick either of the following options)*

This change added tests in `ScalarOperatorsTest` and `SqlExpressionTest`

*(example:)*

- *Added integration tests for end-to-end deployment with large payloads (100MB)*
- *Extended integration test for recovery after master (JobManager) failure*
- *Added test that validates that TaskInfo is transferred only once across recoveries*
- *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency): (yes / no)
- The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
- The serializers: (yes / no / don't know)
- The runtime per-record code paths (performance sensitive): (yes / no / don't know)
- Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / no / don't know)
- The S3 file system connector: (yes / no / don't know)

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by XuQianJin-Stars** [comment]
hi @wuchong  I have modified this PR. Do you have time to review it.
best,
qianjin

File: docs/dev/table/functions.md [hunk 1]
```diff
@@ -1454,28 +1454,16 @@ HEX(string)
...
     <tr>
       <td>
         {% highlight text %}
 TRUNCATE(numeric1, integer2)
 {% endhighlight %}
       </td>
       <td>
-        <p>Returns a <i>numeric</i> after the truncation.</p>
-        <p>E.g. truncate(42.345, 2) to 42.34.</p>
+           <p>Returns a <i>numeric</i> of truncated to <i>integer2</i> decimal places. Returns NULL if <i>numeric1</i> or <i>integer2</i> is NULL.If <i>integer2</i> is 0,the result has no decimal point or fractional part.<i>integer2</i> can be negative to cause <i>integer2</i> digits left of the decimal point of the value to become zero.This function can also pass in only one <i>numeric1</i> parameter and not set <i>Integer2</i> to use.</p>
```
Comment (wuchong) [inline docs/dev/table/functions.md]: Please describe what's the behavior when pass only one argument.

**Comment by XuQianJin-Stars** [comment]
hi @wuchong Ok, let me revise the description 。
best 
qianjin

**Comment by XuQianJin-Stars** [comment]
hi @wuchong I have modified this PR. Do you have time to review it.
best,
qianjin

**Comment by wuchong** [comment]
Thank @XuQianJin-Stars  and sorry for the late reply. The PR looks good to me now. 

Will merge this soon.

**Comment by XuQianJin-Stars** [comment]
hi @wuchong Thank you very much. I know you are very busy.
best 
qianjin

**Pull request closed without merging**